automation/provisioning.go

```diff
@@ -0,0 +1,295 @@
+package automation
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/nutanix-cloud-native/ndb-operator/api/v1alpha1"
+	ndbv1alpha1 "github.com/nutanix-cloud-native/ndb-operator/api/v1alpha1"
+	clientsetv1alpha1 "github.com/nutanix-cloud-native/ndb-operator/automation/clientset/v1alpha1"
+	"github.com/nutanix-cloud-native/ndb-operator/common"
+	"github.com/nutanix-cloud-native/ndb-operator/ndb_api"
+	"github.com/nutanix-cloud-native/ndb-operator/ndb_client"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const namespace_default = "default"
+
+// This function is called from the SetupSuite() function of all testsuites.
+// It loads environment variables, instantiate resources, waits for db to be ready, and pod to start.
+func ProvisioningTestSetup(ctx context.Context, st *SetupTypes, clientset *kubernetes.Clientset, v1alpha1ClientSet *clientsetv1alpha1.V1alpha1Client, t *testing.T) (err error) {
+	logger := GetLogger(ctx)
+	logger.Println("ProvisioningTestSetup() starting...")
+
+	// Nil check
+	if st == nil || clientset == nil || v1alpha1ClientSet == nil {
+		errMsg := "Error: ProvisioningTestSetup() starting ended! "
```

Comment:
Can we use a better statement here instead of `starting ended`?
Perhaps something like `nil parameters/arguments` or `initialization failed` etc?